packages/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+/**
+ An extensible react instance creation delegate. This class will loop through each `ExpoReactDelegateHandler` to determine the winner to create the instance.
+ */
+@objc
+public class ExpoReactDelegate: NSObject {
+  private let handlers: [ExpoReactDelegateHandler]
+
+  public init(handlers: [ExpoReactDelegateHandler]) {
+    self.handlers = handlers
+  }
+
+  @objc
+  public func createBridge(delegate: RCTBridgeDelegate, launchOptions: [AnyHashable : Any]?) -> RCTBridge {
+    self.handlers.forEach { $0.bridgeWillCreate() }
+    let result = self.handlers.lazy
+      .compactMap { $0.createBridge(reactDelegate: self, bridgeDelegate: delegate, launchOptions: launchOptions) }
+      .first(where: { _ in true }) ?? RCTBridge(delegate: delegate, launchOptions: launchOptions)!
+    self.handlers.forEach { $0.bridgeDidCreate(bridge: result) }
+    return result
+  }
+
+  @objc
+  public func createRootView(bridge: RCTBridge, moduleName: String, initialProperties: [AnyHashable : Any]?) -> RCTRootView {
+    return self.handlers.lazy
+      .compactMap { $0.createRootView(reactDelegate: self, bridge: bridge, moduleName: moduleName, initialProperties: initialProperties) }
+      .first(where: { _ in true }) ?? RCTRootView(bridge: bridge, moduleName: moduleName, initialProperties: initialProperties)
+  }
+
+  @objc
+  public func createRootViewController() -> UIViewController {
+    return self.handlers.lazy
+      .compactMap { $0.createRootViewController(reactDelegate: self) }
+      .first(where: { _ in true }) ?? UIViewController()
```

Comment:
should we warn if multiple libraries try to provide their own root view controller? same for root view / bridge

Reply:
i'd like to do it in another pr (probably for next sdk cycle).
  - we may to align the design for showing warnings on android.
  - to implement the warnings, we should introduce new interfaces like `function handlerWillCreateBridge(): boolean`. because we are using `.lazy` here (`.asSequence()` on android), this is to prevent multiple bridges initialized even we just pick one instance returning  from the highest priority handler.